api/integrations/launch_darkly/services.py

```diff
@@ -109,15 +119,199 @@ def _create_tags_from_ld(
+                environment=environment,
+                priority=index,
+            )
+
+            # Enable rules by default. In LD, rules are enabled if the flag is on.
+            FeatureState.objects.update_or_create(
+                feature=feature,
+                feature_segment=feature_segment,
+                environment=environment,
+                defaults={"enabled": True},
+            )
+        return
+
+    segment = Segment.objects.create(name=name, project=project, feature=feature)
+
+    # A parent rule has two children: one for the regular conditions and multiple for the negated conditions
+    # Mathematically, this is equivalent to:
+    # any(X, Y, !Z) = any(any(X,Y), none(Z))
+    # Or with more parameters,
+    # any(X, Y, !Z, !W) = any(any(X,Y), none(Z), none(W))
+    # Here, any(X,Y) is the child rule. none(Z) and none(W) are the negated children.
+    # Since there is no !X operation in Flagsmith, we wrap negated conditions in a none() rule.
+    parent_rule = SegmentRule.objects.create(segment=segment, type=SegmentRule.ANY_RULE)
+    child_rule = SegmentRule.objects.create(rule=parent_rule, type=SegmentRule.ANY_RULE)
+
+    for clause in clauses:
+        operator = _ld_operator_to_flagsmith_operator(clause["op"])
+        value = _convert_ld_value(clause["values"], clause["op"])
+        _property = clause["attribute"]
+
+        if operator is not None:
+            if clause["negate"] is True:
+                negated_child = SegmentRule.objects.create(
+                    rule=parent_rule, type=SegmentRule.NONE_RULE
+                )
+                rule = negated_child
+            else:
+                rule = child_rule
+
+            condition, _ = Condition.objects.update_or_create(
+                rule=rule,
+                property=_property,
+                value=value,
+                operator=operator,
+                created_with_segment=True,
+            )
+            logger.warning("Condition created: " + str(condition))
+        else:
+            logger.warning(
+                "Can't map launch darkly operator: " + clause["op"] + ", skipping..."
+            )
+
+    feature_segment, _ = FeatureSegment.objects.update_or_create(
+        feature=feature,
+        segment=segment,
+        environment=environment,
+    )
+
+    # Enable rules by default. In LD, rules are enabled if the flag is on.
+    FeatureState.objects.update_or_create(
+        feature=feature,
+        feature_segment=feature_segment,
+        environment=environment,
+        defaults={"enabled": True},
+    )
+    return
+
+
+def _import_targets(
+    ld_flag_config: ld_types.FeatureFlagConfig,
+    feature: Feature,
+    environment: Environment,
+) -> None:
+    if "targets" in ld_flag_config:
+        logger.warning("Targets: " + str(ld_flag_config["targets"]))
+
+    if "contextTargets" in ld_flag_config:
+        logger.warning("Context targets: " + str(ld_flag_config["contextTargets"]))
+
```

Comment:
@khvn26 I might need some guidence here. Which objects are storing the variations for individual targets?

Reply:
[Remind myself]: Discuss this in Wednesday's meeting. I kinda understand how this will work, but need to double check with the team as users can't be added to segments individually but rather they can only be added with property & segment rule matches.